### PR TITLE
ID verification rework

### PR DESCRIPTION
**Pull request by blrubenstein**
To update email once it is confirmed working. Don't want to spam that google group until I have it confirmed as functional.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 1]
```diff
@@ -389,19 +390,32 @@ private User initializeUserIfNeeded() {
...
-      new Address()
-        .setStreet1(request.getStreetLine1()).setStreet2(request.getStreetLine2())
-        .setCity(request.getCity()).setSubdivision(request.getState())
-        .setPostalCode(request.getZip()).setCountryCode("US"),
-      request.getDob(),
-      request.getDocumentType(), request.getDocumentNumber()
-    );
+  public ResponseEntity<Profile> submitIdVerification() {
+    User user = userProvider.get();
+    if (user.getRequestedIdVerification() != null) {
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: Shouldn't this be == null? Or == null || == false?

You probably don't want to send more than one of these per user.

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 2]
```diff
@@ -389,19 +390,32 @@ private User initializeUserIfNeeded() {
...
-    Person person = blockscoreService.createPerson(
-      request.getFirstName(), request.getLastName(),
-      new Address()
-        .setStreet1(request.getStreetLine1()).setStreet2(request.getStreetLine2())
-        .setCity(request.getCity()).setSubdivision(request.getState())
-        .setPostalCode(request.getZip()).setCountryCode("US"),
-      request.getDob(),
-      request.getDocumentType(), request.getDocumentNumber()
-    );
+  public ResponseEntity<Profile> submitIdVerification() {
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: Are you separately planning on adding this field to Profile / API so the UI can show when email verification request has been sent?

File: ui/src/app/views/id-verification-page/component.html [hunk 3]
```diff
@@ -1,40 +0,0 @@
...
-  <div>
-    <label for="first-name">First name</label>
-    <input id="first-name" type="text" size="30" autofocus [(ngModel)]="request.firstName">
-  </div>
-  <div>
-    <label for="last-name">Last name</label>
-    <input id="last-name" type="text" size="30" [(ngModel)]="request.lastName">
-  </div>
-  <div>
-    Address<br>
```
Comment (danqrodney) [inline ui/src/app/views/id-verification-page/component.html]: So we're entirely getting rid of the old ID verification stuff? (Are we going to reintroduce something like this later when we integrate with a new vendor?)

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java (hunk 1 above)
Comment (blrubenstein) [reply api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: Correct! That was a mistake. Have fixed.

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 4]
```diff
@@ -389,19 +390,32 @@ private User initializeUserIfNeeded() {
...
+    User user = userProvider.get();
+    if (user.getRequestedIdVerification() == null || user.getRequestedIdVerification() == false) {
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: Thinking about it some more... what happens if somebody enters in some information incompletely or incorrectly, requests a review, and gets rejected... will we clear out this flag at that point so they can fix their info and request verification again?

Comment (blrubenstein) [reply api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: Yeah, my understanding is that we don't have too much process around rejection/re-approval/re-review right now.

**Pull request merged**